client/state/jetpack-connect/reducer/jetpack-auth-attempts.js

```diff
@@ -6,26 +6,29 @@ import { AUTH_ATTEMPS_TTL } from '../constants';
 import { isStale } from '../utils';
 import { JETPACK_CONNECT_COMPLETE_FLOW, JETPACK_CONNECT_RETRY_AUTH } from 'state/action-types';
 import { jetpackAuthAttemptsSchema } from './schema';
+import { keyedReducer } from 'state/utils';
 
-export default function jetpackAuthAttempts( state = {}, action ) {
-	switch ( action.type ) {
+export function authAttempts( state = undefined, { type, attemptNumber } ) {
+	switch ( type ) {
 		case JETPACK_CONNECT_RETRY_AUTH:
-			const slug = action.slug;
-			let currentTimestamp = state[ slug ] ? state[ slug ].timestamp || Date.now() : Date.now();
```

Comment:
This used to cover cases when we have the attempt number but we don't have a timestamp. I'm not sure how we would reach this exact case, and we have no tests for it, which makes me think it's probably unnecessary. Could you just confirm that's the case and we're not removing it intentionally?

Reply:
👍 I don't see any way for state to not be `undefined`, without being and object including `timestamp`